doc/howtos/rdtraining/08_relations.rst

```diff
@@ -0,0 +1,258 @@
+
+In Odoo, there are two models to which we commonly refer to:
+
+- ``res.partner``: a partner is a physical or legal entity. It can be a company, an individual, or
+  even a contact address.
+- ``res.users``: it contains the users of the system. Users can be 'internal': they have
+  access to the Odoo backend. Or they can be 'portal': they cannot access the backend, only the
+  frontend (e.g. to access their previous orders on the eCommerce).
+
+.. exercise:: Add the buyer and the salesperson
+
+    Add a buyer and a salesperson on the ``estate.property`` model using the two common models
+    mentioned. They should be added in a new tab of the form view, as depicted.
+
+    The default value for the salesperson must be the current user. The buyer should not be copied.
+
+    Tip: to get the default value, check the note below or find an example
+    `here <https://github.com/odoo/odoo/blob/5bb8b927524d062be32f92eb326ef64091301de1/addons/crm/models/crm_lead.py#L92>`__.
+
+.. note::
+
+    The object ``self.env`` gives access to request parameters and other useful
+    things:
+
+    - ``self.env.cr`` or ``self._cr`` is the database *cursor* object; it is
+      used for querying the database
+    - ``self.env.uid`` or ``self._uid`` is the current user's database id
+    - ``self.env.user`` is the current user's record
+    - ``self.env.context`` or ``self._context`` is the context dictionary
+    - ``self.env.ref(xml_id)`` returns the record corresponding to an XML id
+    - ``self.env[model_name]`` returns an instance of the given model
+    
+Let's now have a look at other types of links.
+
+Many2many
+=========
+
+**Reference**: the documentation related to this topic can be found in
+:class:`~odoo.fields.Many2many`.
+
+.. note::
+
+    **Goal**: at the end of this section:
+    
+    - a new ``estate.property.tag`` model is created with the corresponding menu and action.
+
+    .. image:: 08_relations/media/property_tag.png
+        :align: center
+        :alt: Property tag
+
+    - Tags are added to the ``estate.property`` model:
+
+    .. image:: 08_relations/media/property_many2many.png
+        :align: center
+        :alt: Property
+
+In our real estate module, we want to define the concept of property tag. A property tag
+is, for example, a property which is 'cozy' or 'renovated'.
+
+A property can have **many** tags, and a tag can be assigned to **many** properties:
+we will use the **many2many** concept.
+
+A many2many is a bidirectional multiple relationship, any record on one side can be related to any
+number of records on the other side.  For example, in order to define a link to the
+``account.tax`` model on our test model, we could write::
+
+    tax_ids = fields.Many2many("account.tax", string="Taxes")
+
+By convention, many2many fields have the ``_ids`` suffix. This means that several taxes can be
+added to our test model. It behaves as a list of records, meaning that accessing the data must be
```

Comment:
maybe here it could be mentioned that a list of records is known as a `recordset`